### ISSUE TITLE
It is possible to use RFC 3389 for dylint lints?

### ISSUE THREAD

**Issue by mondeja** [trailofbits/dylint#1657]
[RFC 3389](https://rust-lang.github.io/rfcs/3389-manifest-lint.html) added a `[lints]` table to be able to configure lints globally from *Cargo.toml*. It's possible to do so with Dylint? It's planned something like that? 

Sorry if this has been discussed somewhere, I can't find any about it.

**Comment by taladar** [comment]
Shouldn't lints in that table be under [lints.dylint] or some similar dylint-specific name if the lints table is used at all?

**Comment by smoelius** [comment]
> Shouldn't lints in that table be under [lints.dylint] or some similar dylint-specific name if the lints table is used at all?

You're asking a good question, @taladar. That is handled by the `cargo` tool. From what I can tell, these are the four `lint` sections it currently [supports](https://github.com/rust-lang/cargo/blob/ebd20d2454fb299834f07b8f5f164783b4e24248/src/cargo/util/toml/mod.rs#L2672):
```rs
        let supported = ["cargo", "clippy", "rust", "rustdoc"];
```
I suspect it would be a tough sell to get a non-`rust-lang` tool in that list.

**Comment by taladar** [comment]
I meant in the context of the earlier experiments with lints.rust that produced issues since Rust does not know the dylint lints and vice versa.

It would probably not such a hard sell to get cargo to ignore the lints.dylint section considering dylint is an unlikely name for any official tool to take? Would there need to be any other support?

**Comment by smoelius** [comment]
> I meant in the context of the earlier experiments with lints.rust that produced issues since Rust does not know the dylint lints and vice versa.

Ah. Without a subtable one gets:
```
{"abs_home_path":"allow"} is not valid under any of the schemas listed in the 'anyOf' keyword
```

> It would probably not such a hard sell to get cargo to ignore the lints.dylint section considering dylint is an unlikely name for any official tool to take? Would there need to be any other support?

Again, valid questions. But suppose Cargo incorporated that change, and tomorrow I archived the Dylint repository. Should the string "dylint" remain in Cargo? We could iterate on this theme 1000 different ways. I've seen similar concerns expressed in Clippy.

TBH, if I were maintaining an official Rust tool like Cargo, I wouldn't want to field requests from third-party tool developers.
